### PR TITLE
plan: call kv.Index Exists for point lookup.

### PR DESCRIPTION
**Pull request by coocood**
Optimization for index point lookup.

### CODE REVIEW

File: plan/plans/index.go [hunk 1]
```diff
@@ -370,20 +385,56 @@ func (r *indexPlan) Next(ctx context.Context) (row *plan.Row, err error) {
...
-		row.RowKeys = append(row.RowKeys, rowKey)
 		return
 	}
 }
 
+func (r *indexPlan) isPointLookup(span *indexSpan) bool {
+	return r.unique && span.lowVal != nil && span.lowVal == span.highVal && !span.lowExclude && !span.highExclude
+}
+
+func (r *indexPlan) lookupRow(ctx context.Context, h int64) (row *plan.Row, err error) {
```
Comment (ngaut) [inline plan/plans/index.go]: Please use unnamed return values.

File: plan/plans/index.go [hunk 2]
```diff
@@ -370,20 +385,56 @@ func (r *indexPlan) Next(ctx context.Context) (row *plan.Row, err error) {
...
+	rowKey := &plan.RowKeyEntry{
+		Tbl: r.src,
+		Key: string(r.src.RecordKey(h, nil)),
+	}
+	row.RowKeys = append(row.RowKeys, rowKey)
+	return
+}
+
+// pointLookup do not seek index but call Exists method to get a handle, which is cheaper.
+func (r *indexPlan) pointLookup(ctx context.Context, val interface{}) (row *plan.Row, err error) {
```
Comment (ngaut) [inline plan/plans/index.go]: Please use unnamed return values.

**Comment by siddontang** [comment]
no test?

**Comment by coocood** [comment]
@siddontang It is covered in tidb_test

**Comment by ngaut** [comment]
LGTM

**Comment by disksing** [comment]
LGTM

**Pull request merged**